pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtimetoofflinesegments/RealtimeToOfflineSegmentsTaskGenerator.java

```diff
@@ -305,7 +305,10 @@ private long getWatermarkMs(String realtimeTableName, List<SegmentZKMetadata> co
       // Find the smallest time from all segments
       long minStartTimeMs = Long.MAX_VALUE;
       for (SegmentZKMetadata segmentZKMetadata : completedSegmentsZKMetadata) {
-        minStartTimeMs = Math.min(minStartTimeMs, segmentZKMetadata.getStartTimeMs());
+        long startTimeMs = segmentZKMetadata.getStartTimeMs();
```

Comment:
Will be more robust if we exclude segments that fails `TimeUtils.timeValueInValidRange()`